modules/renter/contractor/dependencies.go

```diff
@@ -0,0 +1,106 @@
+package contractor
+
+import (
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// These interfaces define the HostDB's dependencies. Using the smallest
+// interface possible makes it easier to mock these dependencies in testing.
+type (
+	consensusSet interface {
+		ConsensusSetPersistentSubscribe(modules.ConsensusSetSubscriber, modules.ConsensusChangeID) error
+	}
+	// in order to restrict the modules.TransactionBuilder interface, we must
+	// provide a shim to bridge the gap between modules.Wallet and
+	// transactionBuilder.
+	walletShim interface {
+		NextAddress() (types.UnlockConditions, error)
+		StartTransaction() modules.TransactionBuilder
+	}
+	wallet interface {
+		NextAddress() (types.UnlockConditions, error)
+		StartTransaction() transactionBuilder
+	}
+	transactionBuilder interface {
+		AddArbitraryData([]byte) uint64
+		AddFileContract(types.FileContract) uint64
+		Drop()
+		FundSiacoins(types.Currency) error
+		Sign(bool) ([]types.Transaction, error)
+		View() (types.Transaction, []types.Transaction)
+	}
+	transactionPool interface {
+		AcceptTransactionSet([]types.Transaction) error
+	}
+
+	hostDB interface {
+		Host(modules.NetAddress) (modules.HostSettings, bool)
+		RandomHosts(n int, exclude []modules.NetAddress) []modules.HostSettings
+	}
+
+	dialer interface {
+		DialTimeout(modules.NetAddress, time.Duration) (net.Conn, error)
+	}
+
+	persister interface {
+		save(contractorPersist) error
+		load(*contractorPersist) error
+	}
+
+	logger interface {
+		Println(...interface{})
+	}
+)
+
+// because wallet is not directly compatible with modules.Wallet (wrong
+// type signature for StartTransaction), we must provide a bridge type.
+type walletBridge struct {
+	w walletShim
+}
```

Comment:
I still find the bridge and shim stuff confusing. Maybe there is a better way? Though I don't have many ideas. again, non-blocking